packages/TorneloScoresheet/src/components/Pin/Pin.tsx

```diff
@@ -9,11 +9,13 @@ import {
 } from 'react-native-confirmation-code-field';
 import PrimaryButton from '../PrimaryButton/PrimaryButton';
 import PrimaryText from '../PrimaryText/PrimaryText';
+import { useError } from '../../context/ErrorContext';
+import { pinValid } from '../../util/arbiterPin';
 
 const CELL_COUNT = 4;
 
 export type PinProps = {
-  onPress: () => void;
+  onPress: (pinCorrect: boolean) => void;
```

Comment:
I think we should only be calling a function if the pin is correct right?
I might be missing something though... but you are never calling onPress(false) in the code
If we never need to let the page containing the pin know that the pin was incorect we should rename this to  onCorrectPinEntered()

Reply:
We also call a function if the pin is false because it reloads the page

<img width="344" alt="image" src="https://user-images.githubusercontent.com/64511606/182517591-888dc292-6d62-4d9e-95f2-74e1c13cedbf.png">
<img width="347" alt="image" src="https://user-images.githubusercontent.com/64511606/182517685-45721f38-8064-4d1d-9682-cade33da26f5.png">

Reply:
so onPress we are calling handleVerify, and if the pin is valid it moves into arbiter mode but if it's incorrect it reloads the pin input so that the user can re-type

Reply:
hmm oh yeh I didn't see that, and is there any reason we are doing that?
Why is the reset pin comming from the parent component shouldn't the Pin component handle all the pin stuff by itself (ie the pinReset field should be a useState from within the Pin component)
Im bringing this up becuase the pin component is not only used ffrom the toolbar, its also used in the lock app thingy if the user switches out of the app and we might end up using it in other places too, so it would be better if the Pin compoment works all by itself with no need for extra inputs except the function to call once the pin intered is correct

Reply:
I hadn't realised it was used in other places sorry, I updated it so that everything happens in the pin component.